src/queries/index.js

```diff
@@ -5,16 +5,38 @@ const defaultFetchPolicy = CozyClient.fetchPolicies.olderThan(
   DEFAULT_CACHE_TIMEOUT_QUERIES
 )
 
-export const buildTimetableQuery = (start, end) => ({
+export const buildAccountsQuery = () => ({
+  definition: () =>
+    Q('io.cozy.accounts')
+      .where({
+        account_type: 'pronote'
+      })
+      .indexFields(['account_type']),
+  options: {
+    as: 'io.cozy.accounts/pronote',
+    fetchPolicy: defaultFetchPolicy
+  }
+})
+
+export const buildTimetableQuery = (sourceAccountIdentifier, start, end) => ({
   definition: () =>
     Q('io.cozy.calendar.events')
       .where({
         start: start ? { $gte: start } : { $gt: null },
-        end: end ? { $lte: end } : { $lt: null }
+        end: end ? { $lte: end } : { $lt: null },
+        cozyMetadata: {
+          sourceAccountIdentifier
+        }
       })
-      .indexFields(['start', 'end']),
+      .indexFields(['start', 'end', 'cozyMetadata.sourceAccountIdentifier']),
```

Comment:
I have a suggestion for this query: 
1. move  'cozyMetadata.sourceAccountIdentifier' in first position in the indexFields array
2. instead of indexing both `start` and `end` fields, only use start. And in the `where` selector, replace with `start:  end ? { $lte: end } : { $lt: null }`. So when you specify a time range, e.g. "{start: 2024-01-01, end: 2024-07-01}` it will only rely on the `start` attribute. Which makes sense, because if you have an event that starts the 2024-07-01 at 23:00 and ends the next day, you should display it, which is currently not the case.

And, as a nice side-effect of 1. + 2., you now have a very efficient index, because you have an equality operator on the first indexed attribute (sourceAccountIdentifier) and a range operator on the second (start). If you reverse the order, or if you have 2 range operators, the indexing will not be efficient. See https://docs.cozy.io/en/tutorials/data/advanced/#indexes-performances-and-design for more insights

Reply:
I changed your code since it was trying to get all the lessons prior to the end date regardless of the start date, I don't know if it's the right syntax but i added the start to it :
```js
start: end ? { $lte: end, $gte: start } : { $lt: null, $gte: start }
```

Commit change -> https://github.com/cozy/cozy-papillon-app/pull/29/commits/76fee8c147e593ba7b440711efb93d32b337a8fa